### DISCUSSION TITLE
Did sessions refresh automatically?

### DISCUSSION THREAD

**Discussion by meotimdihia** [ory/kratos#2694] [Q&A]
Hi,
Did sessions refresh automatically? The documentation doesn't mention this problem: 
https://www.ory.sh/docs/kratos/session-management/refresh-extend-sessions

**Comment by meotimdihia** [comment]
@aeneasr please help me with this problem.

**Comment by joekrill** [comment]
I am wondering the same thing. Reading through the documentation, it initially seem to be saying that you must manually extend the session using the `adminRefreshSession` endpoint. But then this warning is what really made me question that interpretation:

> If you set earliest_possible_extend to lifespan, all sessions will constantly be refreshed!

That makes it seem like the session will be extended automatically on _any_ request? Would be great to get some clarification on the behavior here.
